### PR TITLE
World in Data : capstone project

### PR DESCRIPTION
**Pull request by Hisoka37**
# Hello CR 😄

 

## In this PR I : 
 - Choose an open API that allows me to get numeric data out of it.
 - Followed design guidelines given by the curriculum.
 - Displayed all the countries on the home page found in the API.
 - Implemented a details page showing detailed information about the clicked country.
 - Added a search bar to search for a desired country from the list.
 - Implemented a function to return back to the home page using the `<` button.
 - Added a test for the pure function in the project.
 - Used react-router for routing through the pages.
 -  Deployed the webpage on `render.com.`

Cheers and Happy coding 😃

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+
+If you like this project kindly send to me an e-mail expressing it, it would make my day and fuel my motivation.
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+
+## 🙏 Acknowledgments <a name="acknowledgements"></a>
+
+I would like to thank Microverse team for this journey.
+
```
Comment (pinkmoon25) [inline README.md]: - [ ] hi, you have done really well with the project, :clap: , Just a few things to fix and it should be completed, first please notice that it is mentioned in the project requirements **The [Creative Commons license of the design](https://creativecommons.org/licenses/by-nc/4.0/) requires that you give appropriate credit to the author. Therefore, you must do it in the README of your project.** So in your acknowledgments section  please provide the appropriate credit to **Original design idea by [Nelson Sakwa on Behance](https://www.behance.net/sakwadesignstudio).**

File: src/tests/componenet.test.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  });
+  test('Snapshot test for Country', () => {
+    const countryDipslayed = render(<Provider store={store}><Country /></Provider>);
+    expect(countryDipslayed).toMatchSnapshot();
+  });
+  test('Snapshot test for Search', () => {
+    const countrySearched = render(<Provider store={store}><Search /></Provider>);
+    expect(countrySearched).toMatchSnapshot();
+  });
+});
```
Comment (pinkmoon25) [inline src/tests/componenet.test.js:22]: - [ ] Again, everything is working well, however I want to mention that testing is also a crucial part in development, currently you have written test cases for components only, as mentioned in the project requirements :point_down: 
image

Please try to add test cases for reducers and actions, I will try to provide some steps you can take below

- Import the necessary dependencies for testing: redux-mock-store, thunk, and jest.
- Create a mock store using the redux-mock-store library.
- Import the reducer and the reducer action creators to be tested.

example snippet : 
import dependencies needed

```
const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

describe('countriesSlice reducers', () => {
  let store;

  beforeEach(() => {
    store = mockStore({
      loading: false,
      countriesData: [],
      countrySearched: [],
      error: false,
      searchTerm: '',
      success: false,
      message: '',
    });
  });

  it('should set the search term', () => {
    const searchTerm = 'test';
    const expectedAction = {
      type: 'countries/setSearchTerm',
      payload: searchTerm,
    };

    store.dispatch(setSearchTerm(searchTerm));
    expect(store.getActions()).toEqual([expectedAction]);
  });
});

```

File: src/components/country/country.css [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+.country-content p span {
+  font-weight: 300;
+}
+
+.countryheader {
+  display: flex;
+  align-items: center;
+  gap: 10px;
+}
```
Comment (pinkmoon25) [inline src/components/country/country.css:45]: - [ ] how about we move this country name and the arrow to the top and make sure the arrow is on top right corner to match the original design layout
currently 
image

design 
image

notice the arrow is on top right :+1:

**Review by pinkmoon25 — CHANGES_REQUESTED** [review]
Hi @Hisoka37 , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights
Linter checks :heavy_check_mark: 
gitflow :heavy_check_mark: 
Metrics API :heavy_check_mark: 
React SPA :heavy_check_mark: 
Redux store :heavy_check_mark: 
Home page and Details page :heavy_check_mark: 
Project deployed :heavy_check_mark: 

- [ ] Good job with the video presentation :+1: , In order to make it more professional, I would advise to showcase bits of your code and how you implemented a certain functionality :pray: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by amejid — APPROVED** [review]
Hi @Hisoka37, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights
- Linter checks passing :heavy_check_mark: 
- Good UI design :heavy_check_mark: 
- Navigation with React router :heavy_check_mark: 
- Filtering working well :heavy_check_mark: 
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**